29 gamecontroller

stephensmeal opened the pull request:
Implemented GameController and created a Unit Test. I don't understand the total functionality of GameController but this is what I have so far. I'm sure it will have to be addressed further.

stephensmeal committed:
Implementation and Unit Tests for GameController Class

yo09975 commented on src/gamecontroller.py:
```diff
@@ -9,20 +15,40 @@ class GameController(object):
+        """Returns MovementEngine reference"""
+        return self._move_engine
+
+    def read_message(self, message: Message):
```
GC should invoke the read_message from the ServerNetworkInterface with the uuid of the player you want the message from.

yo09975 replied:
We still need to define the "payload" of each message, so that not only can the GC "route" the message to the appropriate subsystem, but it can call the appropriate state changes.

patmay committed:
Add toggle example and better error handling in Button.

patmay committed:
Implement initial picker functionality.

patmay committed:
Finish picker and create picker example.

patmay committed:
Merge pull request #100 from yo09975/46_picker

46 picker

patmay committed:
Implement Dialog, and example, and add visbility to View.

patmay committed:
Merge pull request #102 from yo09975/43_dialog

Implement Dialog, and example, and add visbility to View.

stephensmeal committed:
Updated Still In Progress GameController Implementation

gamecontroller.py - Updated implementation
gamestate.py - Bug fix in start() to set correct first player
gamestatus.py - Added and remove GameStatus(es) for functionality

stephensmeal committed:
Bug Fix In GameState And Fix Unit Tests

Previous change to gamestate.py start() method needed to be
fixed to call self.next_turn(). GameState unit test
(test_gamestate.py) needed to be changed to reflect updated
GameStatus(es).

patmay committed:
Implement SuggestionDialog. Requried changes to other View classes.

patmay committed:
Update UI examples so they work properly with View changes.

patmay committed:
Merge pull request #106 from yo09975/54_sugg_dialog

54 sugg dialog

stephensmeal committed:
Incremental GameController and Test Implementation

stephensmeal committed:
Implementation of GameController and Bug Fixes

gamecontroller.py - Addresses Comments and Added Helper Functions
suggest_engine.py - Added is_valid_suggestion() function to ensure
    suggesting player is in suggested room
test_gamecontroller.py - Removed from branch to increase cycle time

stephensmeal committed:
Fixed Pep8 Issues in GameController

gamecontroller.py - Fixed Pep8 Issues

yo09975 commented:
Duplicate PR once the base of PR 104 is changed to integ.
